src/docs/arch-hs.dox

```diff
@@ -4,5 +4,111 @@ The History Store in WiredTiger tracks old (all but the latest committed)
+WiredTiger uses a standard row-store table to implement History Store internally.
+The key is a combination of:
+- \c btree ID of user table this update belongs to
+- \c record key (byte-string for row-store, record number for column-store)
+- \c start timestamp for the update
+- \c counter value
+This key format above allows us to search efficiently for a given record key and read
+timestamp combination. The last part of the key is a monotonically increasing counter
+to keep the key unique in scenarios where we have multiple transactions committing at
+the same timestamp. As the key for the History Store table is different for row- and column-store,
+we store both keytypes in a WT_ITEM because otherwise we'd need two history store files with
+different key formats. We can make the History Store table's key standard by moving the source
+key into the History Store table value, but that doesn't make the coding any simpler, and it makes
+the History Store table's value more likely to overflow the page size when the row-store key is
+relatively large.
+
+The corresponding value for each key is stored as a combination of:
+- \c stop timestamp
+- \c durable timestamp
+- \c update type
+- \c value.
+
+Note that every update in the Hisotry Store has a valid stop timestamp and transaction associated
+with it, except for cases where the update preceded a prepared update. The stop timestamp
+is fixed in this case once prepared update is resolved.
+
+@section arch_hs_initialize Historty Store initialization
+WiredTiger checks for an existing History Store table at startup and creates a new table if it is
+not able to find one. History Store table is usually created when WiredTiger creates a new database,
+with one exception where WiredTiger is openning an existing database created by an older version of
+WiredTiger that didn't support Histoty Store feature. WiredTiger uses \c WiredTigerHS.wt as the
+filename for the History Stable table.
+
+@section arch_hs_cursor Historty Store cursor interface
+WiredTiger uses a cursor interface for History Store table to make it easier for different modules
+to perform data operations on the History Strore table. A new cursor can be opened by calling
+\c __wt_curhs_open(). History Store cursor implementation supports most of the standard \c WT_CURSOR
+methods, with one key exception being \c WT_CURSOR::search() method. Recall that the key contains a
+counter value and it is expected that caller can not possibly know the correct counter value to use.
+Therefore, \c WT_CURSOR::search_near() is used for all search operations. Two helper functions,
+\c __wt_curhs_search_near_before() and \c __wt_curhs_search_near_after(), have been provided to
+facilitate searching the required record in the History Store table.
+
+@section arch_hs_cursor_visibilty Historty Store cursor interface and visibility rules
+When using the Historty Store cursor interface, the user can configure the type of visibility checks
```

Comment:
I think better phrasing would be "the user can configure the isolation level used by the cursor for reading history store records."

Reply:
I suppose it isn't a true isolation level, but I'm not sure if the reader would be familiar with the terminology "visibility checks"

Reply:
I believe if you are reading this section you are probably someone from WiredTiger team who is trying to use history store  cursor. As such, I'll expect them to know something about visibility checks and isolation levels.